protocols/protocol.py

```diff
@@ -27,6 +26,37 @@ def default(self, obj):
         return ret
 
 
+class ValidationResult(object):
+
+    def __init__(self, result=None, messages=None):
+        self.result = result or True
+        self.messages = messages or []
+        self.msg = "Class: {class_name} expects field: {field_name} "
+        self.msg += "with schema type: {schema_type} but received value: {value}"
+        self.schema_type_msg = "Schema: {expected_schema} has type: {schema_type} "
+        self.schema_type_msg += "but received datum: {datum}"
```

Comment:
The message may be clearer if you enclose the variable fields between brackets instead of writing them after a colon. It is not clear for the reader where the value ends. This would also help parsing if necessary downstream.